AssemblyProcessor multi-targeting fix

Ethereal77 opened the pull request:
# PR Details

This PR modifies the `AssemblyProcessor` project to build for both `net6.0` and `netstandard2.0`, with the intention of allowing both the .NET Framework version of MSBuild (like the one used by Visual Studio) and the .NET 5+ version (like the one invoked by `dotnet build` or `dotnet msbuild`) to load the correct version of the Assembly Processor assembly.

## Description

* `AssemblyProcessor` now builds `net6.0` and `netstandard2.0`.
* Both versions are copied to `deps` to their corresponding subdirectory.
* Modified `Stride.Core.targets` to select the correct assembly based on the framework of MSBuild.
* Made GameStudio always reference the `net6.0` (or newer) variant of the Assembly Processor. This is because GameStudio is not multi-targeting anymore, so there's no need to choose.
* Updated `Mono.Cecil` and made its version the same for both `Stride.Core.AssemblyProcessor` and `Stride.GameStudio`.
* Fixed the Assembly Processor test project. It was not compiling correctly due to missing `Xunit` reference.

## Related Issue

This PR is based primarily in the PR [#1480](https://github.com/stride3d/stride/pull/1480) by [ericwj](https://github.com/ericwj), but updated based on a recent conversation on Discord.
The original issue that motivated that PR was #1479.

## Motivation and Context

The Assembly Processor assemblies in `deps` were stale as the project wasn't compiling and updating the `netstandard2.0` version anymore, which is the one used by the MSBuild invocations used by Visual Studio.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

ericwj committed:
Fix asset compiler build.

ericwj committed:
Fix access violation in particle system simulation processor.

ericwj committed:
Correct exception behavior of derived `System.IO.Stream` types.
Remove manually copied documentation comments in favor of `<inheritdoc/>`.

ericwj committed:
Do not use trailing alignment padding for source parameter collection buffer.
Fixes pre-existing buffer overrun and those introduced by assuming trailing alignment padding.

ericwj committed:
Fix whitespace in LZ4Stream.cs

ericwj committed:
Rewrite parameter buffer copying.
To account for differences in type/size of equally named parameters.

ericwj committed:
`StreamExtensions` should have been public.

Basewq committed:
[Samples] Fix TopDownRPG punch animation caused by breaking change in raycast from stride3d#1105

ericwj committed:
Fix debug assertion error in LZ4Stream.Write.

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

ericwj committed:
Typo.

ericwj committed:
Obsolete `SerializationStream.NativeStream` to change the name.

VaclavElias committed:
Readme: Build Instructions improvements (#1494)

ericwj committed:
Comments and whitespace.

palkercode committed:
Added overload for Entity constructor

insomnyawolf committed:
[Debug] DebugTextSystem configurable time on-screen (#1515)

garychia committed:
Enable multiple selections of the axes of grids to be drawn

Basewq committed:
[Build] Add build support to referencing projects that use package version with Semantic Versioning 2.0.0 format

Basewq committed:
[Math] Fix Matrix.Invert to return zero matrix if inversion fails instead of NaNs, caused by breaking change from (#1301) (#1520)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

manio143 committed:
Merge pull request #1521 from Basewq/semver-build-support

Add build support to referencing projects that use SemVer 2.0.0 format

palkercode committed:
Merge branch 'stride3d:master' into master

arturoc committed:
openxr: Add static extensionsvariable

To allow to set extensions from outside the engine

Basewq committed:
[Math] Fix Matrix.Decompose, Matrix.DecomposeXYZ & Quaternion.RotationYawPitchRoll to better handle the edge cases where the pitch (or yaw) is +/-90 degrees

Basewq committed:
[Editor] Fix rotation value showing -0 due to MathF respecting negative sign even at zero

manio143 committed:
Configure Samples.Tests and SamplesTestServer to auto resolve packages correctly

manio143 committed:
Merge pull request #1535 from manio143/SamplesTestsAutoResolver

Fix Samples.Tests and SamplesTestServer to auto-resolve packages

Eideren committed:
[Template] Fix first and third person delta time usage (#1511)

Eideren committed:
[Template] FPS, remove unused logic (#1512)

Eideren committed:
Addressing PVS report

Eideren committed:
Strip encoding issues

Eideren committed:
Replace encoding issue

manio143 committed:
Improve documentation of ListFiles and add tests for DatabaseFileProvider implementation

manio143 committed:
Replace string concatenation with interpolation

manio143 committed:
Merge pull request #1547 from manio143/dbFileProviderSearchDocumentation

Improve documentation of ListFiles and add tests for DatabaseFileProvider implementation

arturoc committed:
OpenXR: Fix Non DirectX11 platforms

manio143 committed:
Merge pull request #1469 from ericwj/MigrateInterop2

Improve memory management, embrace modern .NET

Jklawreszuk committed:
Replace native ThreadSleep with .NET Thread.Sleep

manio143 committed:
Merge pull request #1523 from Jklawreszuk/replace-native-sleep-with-net-thread-sleep

Replace native ThreadSleep with .NET Thread.Sleep

Ethereal77 committed:
Use Path.Combine() to concatenate MSBuild paths

manio143 committed:
Merge pull request #1569 from Ethereal77/fix-msbuild-path-combine

Use Path.Combine() to concatenate MSBuild paths

arturoc committed:
Merge branch 'openxr-dx11-master' of github.com:arturoc/stride into openxr-dx11-master

arturoc committed:
VRRendererSettings: Remove CopyMirror property code. Not really useful since this can only be properly evaluated at runtime

arturoc committed:
Update Silk OpenXR nuget

arturoc committed:
VirtualReality project: change OpenXR Silk to 2.10.1 to match other Silk packages in the rest of the engine

manio143 committed:
Merge pull request #1545 from Eideren/addressing_pvs_report

[Misc] Addressing PVS report

dependabot[bot] committed:
Bump System.Drawing.Common from 4.7.0 to 4.7.2 in /sources/engine/Stride

Bumps [System.Drawing.Common](https://github.com/dotnet/corefx) from 4.7.0 to 4.7.2.
- [Release notes](https://github.com/dotnet/corefx/releases)
- [Commits](https://github.com/dotnet/corefx/commits)

---
updated-dependencies:
- dependency-name: System.Drawing.Common
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

manio143 committed:
Merge pull request #1560 from stride3d/dependabot/nuget/sources/engine/Stride/System.Drawing.Common-4.7.2

Bump System.Drawing.Common from 4.7.0 to 4.7.2 in /sources/engine/Stride

palkercode committed:
Merge branch 'stride3d:master' into master

garychia committed:
Simplify code

palkercode committed:
Update Entity.cs

Updated with Eideren's suggestion

manio143 committed:
Merge pull request #1499 from palkercode/master

Added overload for Entity constructor

Brian-Hamlin-BCS committed:
Updated Entity.cs

Fixed an error preventing the solution from compiling due to Entity.cs.

manio143 committed:
Merge pull request #1578 from Brian-Hamlin-BCS/master

Updated Entity.cs

skohanowski committed:
Added deconstructors to most of the structs in Stride.Mathematics

skohanowski committed:
Addressed @manio143 feedback regarding deconstructors.

manio143 committed:
Merge pull request #1582 from skohanowski/Deconstructors

Added deconstructors to most of the structs in Stride.Mathematics

manio143 committed:
Make a few changes around viewport grid axis rotation and hiding

manio143 committed:
Introduce an enum to avoid allocating the nullable object

garychia committed:
Fix the default scene settings data

mattiascibien committed:
Fix documentation error on Int2

manio143 committed:
Merge pull request #1585 from mattiascibien/fix_int2_docs

Fix documentation error on Int2

Jklawreszuk committed:
Use LZ4 as external dependency

manio143 committed:
Merge pull request #1566 from Jklawreszuk/lz4-package

Use LZ4 as external dependency

Jklawreszuk committed:
Remove settings.json and update gitignore

manio143 committed:
Merge pull request #1586 from Jklawreszuk/vscode-gitignore

Remove settings.json and update gitignore

Eideren committed:
[Rendering] Addressing PVS report - SDL (#1546)

Eideren committed:
[Physics] Update WorldMatrix when forcing transformation on PhysicsComponent (#1406)

xen2 committed:
Merge remote-tracking branch 'arturoc/openxr-dx11-master' (closes #1483)

mattiascibien committed:
Fix #1466: adding scripts by dragging

Jklawreszuk committed:
Cleanup Windows Phone refs

Jklawreszuk committed:
Cleanup Silverlight refs

Jklawreszuk committed:
Cleanup WinStore refs

Jklawreszuk committed:
Cleanup Win8 & Win10 refs

Jklawreszuk committed:
Cleanup NET Portable,MicroFramework,Client

Jklawreszuk committed:
Remove WinStore content for VRSandbox sample

azeno committed:
[Graphics] If possible use flip model for swap chain

https://devblogs.microsoft.com/directx/dxgi-flip-model

Jklawreszuk committed:
Move launcher prerequisites validation to separate file

xen2 committed:
Keep local copy of signclient to avoid issues with teamcity

Jklawreszuk committed:
Launcher - Remove unnecessary namespaces, format and simplify code

manio143 committed:
Merge pull request #1605 from Jklawreszuk/launcher-refactor

[Launcher] Code refactor

garychia committed:
Merge pull request #1 from manio143/viewportgrid_changes

Viewportgrid changes

manio143 committed:
Merge pull request #1516 from garychia/GridsInMultiplePlanes

Enable multiple selections of the axes of grids to be drawn

manio143 committed:
Document what's in the Core folder

manio143 committed:
Merge pull request #1610 from manio143/document_core_folder

[Docs] Add a Readme with information on what's in the Core folder

manio143 committed:
Merge pull request #1588 from mattiascibien/fix_adding_scripts_by_dragging

Fix #1466: adding scripts by dragging

manio143 committed:
Update NuGet version to cover most .NET 6 SDK users for MSBuild issue

manio143 committed:
Add debug information & remove legacy code

manio143 committed:
Merge pull request #1613 from manio143/fix_script_editor

Update NuGet dependencies to 6.4.0 (temporary fix for ScriptEditor issues)

dependabot[bot] committed:
Bump SSH.NET from 2016.1.0 to 2020.0.2 in /sources/engine/Stride.Assets

Bumps [SSH.NET](https://github.com/sshnet/SSH.NET) from 2016.1.0 to 2020.0.2.
- [Release notes](https://github.com/sshnet/SSH.NET/releases)
- [Commits](https://github.com/sshnet/SSH.NET/compare/2016.1.0...2020.0.2)

---
updated-dependencies:
- dependency-name: SSH.NET
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Doprez committed:
fixed constructors in CsharpBeginner template

manio143 committed:
Merge pull request #1618 from Doprez/fix-begginner-errors

fixed constructors in CsharpBeginner template

manio143 committed:
Merge pull request #1614 from stride3d/dependabot/nuget/sources/engine/Stride.Assets/SSH.NET-2020.0.2

Bump SSH.NET from 2016.1.0 to 2020.0.2 in /sources/engine/Stride.Assets

manio143 committed:
Merge pull request #1428 from vvvv/dev/azeno/native-libs-use-runtimes-folder

[Build] Native libraries get copied to runtimes/RID/native folder

azeno committed:
[Graphics] Fixes Vsync off when using new flip model

azeno committed:
[Build] Fixes native libraries not copied to output folder of unit tests

For example the Stride.Navigation.Tests didn't contain the necessary libnavigation.dll in its runtimes folder.

xen2 committed:
Merge pull request #1632 from stride3d/dev/azeno/unit-test-native-lib-load-fix

[Build] Fixes native libraries not copied to output folder of unit tests

azeno committed:
[Graphics] Adds try/catch blocks to flip model and tearing support queries to support older OS

azeno committed:
[Graphics] Fixes crash when flip model is not supported

The `AllowTearing` flag must not be set in that case

Doprez committed:
rebased branch

manio143 committed:
Merge pull request #1643 from Doprez/change-msbuild

rebased branch

manio143 committed:
Merge pull request #1591 from Jklawreszuk/cleanup-legacy-platforms

Cleanup legacy platforms

NicusorN5 committed:
Possible fix for the animation importer crashing

`FbxTime::GetOneFrameValue(FbxTime::eFrames1000);` returns the smallest value, so there's the highest animation resolution possible(?)

Jklawreszuk committed:
[Core] Remove some obsolete code

NicusorN5 committed:
Update AnimationConverter.h

NicusorN5 committed:
Also check negative values

manio143 committed:
Merge pull request #1653 from NicusorN5/master

Possible fix for the animation importer crashing

NicusorN5 committed:
Update the forum link with Github discussions link

Doprez committed:
added null check

manio143 committed:
Merge pull request #1656 from Doprez/fix-NativeDevice-causing-crash

fix Crash when NativeDeviceContext is null

azeno committed:
[Build] Fixes location of native openxr_loader.dll in Stride.VirtualReality package as well as used Silk version

This broke with the merge of PR #1428 because these lines of code where added after #1428 was created.

xen2 committed:
Merge pull request #1660 from vvvv/fix-openxr-load-location

[Build] Fixes location of native openxr_loader.dll in Stride.VirtualReality package

manio143 committed:
Merge pull request #1654 from Jklawreszuk/core-obsoletes

[Core] Remove some obsolete code

Jklawreszuk committed:
Fix XML warnings (#1678)

Jklawreszuk committed:
[deps] - Remove unused ATITC folder (#1668)

Jklawreszuk committed:
[deps] - Remove unused HtmlRenderer (#1670)

Jklawreszuk committed:
[deps] - Remove unused ExpresionBlend Sdk (#1669)

Jklawreszuk committed:
Remove VisualStudio reattach deadcode (#1677)

Jklawreszuk committed:
Fix XML documentation (#1680)

IXLLEGACYIXL committed:
fix argument reference

IXLLEGACYIXL committed:
remvoe reference to non existing class that doesnt get referenced

IXLLEGACYIXL committed:
fix light to renderlight

IXLLEGACYIXL committed:
blend layer context is the only class that actually sets and evaluates the enum values

IXLLEGACYIXL committed:
remove duplicate through inheritdoc

IXLLEGACYIXL committed:
add missing inheritdoc

IXLLEGACYIXL committed:
add missing inheritdoc

IXLLEGACYIXL committed:
fix wrong reference to draweffect parameters

IXLLEGACYIXL committed:
Merge branch 'FixXmlDocs' of https://github.com/NexStandard/stride into FixXmlDocs

IXLLEGACYIXL committed:
fix wrong comment. False class in  comment

IXLLEGACYIXL committed:
duplicate comment

IXLLEGACYIXL committed:
duplicate comment

IXLLEGACYIXL committed:
duplicate comment

IXLLEGACYIXL committed:
fix wrong interface link

IXLLEGACYIXL committed:
remove duplicate

IXLLEGACYIXL committed:
fix wrong reference

IXLLEGACYIXL committed:
fix resourcegrouplayout reference

IXLLEGACYIXL committed:
fix vertex comment

IXLLEGACYIXL committed:
wrong param name, its still a weird name as it was previous a sorter i assume.

IXLLEGACYIXL committed:
ParticleSorter is now ParticleList from previous interface renaming

manio143 committed:
Merge pull request #1682 from NexStandard/FixXmlDocs

[Docs] Fix xml docs

IXLLEGACYIXL committed:
[Docs] Fix xml docs (#1683)

Ethereal77 committed:
Change AssemblyProcessor to multi TFM for netstandard2.0

Ethereal77 committed:
Reference the correct AssemblyProcessor in Stride.Core.targets

* It will decide which one based on the version of MSBuild.

Ethereal77 committed:
Change GameStudio to always use the same TFM as Stride.Core

* As GameStudio is not multi-targeting anymore, a single TFM should be safe.
* Updated Mono.Cecil to match the one used by AssemblyProcessor.

Ethereal77 committed:
Removed unnecessary net472 TFM from AssemblyProcessor

Ethereal77 committed:
Remove old version of AssemblyProcessor from deps

Ethereal77 committed:
Fix the AssemblyProcessor tests that were not compiling

Ethereal77 commented:
@dotnet-policy-service agree

Ethereal77 commented:
A question about one of the commits:

As of this PR `Stride.Core.AssemblyProcessor` can be built both for `net6.0` and `netstandard2.0` and it will be copied to `deps`.
Should we delete the already compiled version we have in `deps\AssemblyProcessor` and instead compile it as a first step so it's fresh and up to date?

* If we **should** delete it (I've already done it in a commit, but can be undone), maybe it should be noted somewhere so people building Stride know it's a requirement to build first the Assembly Processor prior to building the main solution.
* If we **shouldn't** delete it, I can commit the updated assemblies in `deps`.

tebjan commented:
from my understanding, the .netstandard2.0 could also just be dropped/deleted. With .NET 8 around the corner and the small community of contributors, the less complexity we have the better.

Ethereal77 commented:
I'd tend to agree. But as I've written in the PR, I think `netstandard2.0` is still needed by the .NET Framework version of MSBuild that Visual Studio uses. Whenever I've tried to use just a .NET Core `AssemblyProcessor`, it has compiled with `dotnet build` and `dotnet msbuild`, but failed in Visual Studio with messages like _"Can't resolve assembly reference System.Runtime 6.0"_ or something like that.
The only reason it built up until now with just `net6.0` is because there was a stale version of `netstandard2.0` in the `deps` directory.

If / when Visual Studio can load .NET Core tasks in MSBuild, or replaces its version of MSBuild with a .NET Core one, then we can say goodbye to `netstandard` for good.

Ethereal77 committed:
Renamed build scripts in solution files and removed missing ones (#1696)

Eideren commented:
@Ethereal77 regarding your question, we could ask users to build that one first in the worst case scenario ...
I understand that library has to process the other solutions to build them properly so I could see how that would be harder than expected, still, is there no way we could build it first as part of building the game studio / the whole solution ?

azeno committed:
[Reflection] Fixes loading of mixed mode assemblies in .NET Core (#1691)

It seems .NET Core behaves slightly different in this regard, throwing a different kind of exception, so previous code no longer applies, even though it's the same assembly being loaded (irrKlang.NET4.dll)

Jklawreszuk committed:
[deps] - Remove unused SSH (#1687)

Jklawreszuk committed:
[deps] Remove unused libgit2 dll's (#1686)

Jklawreszuk committed:
[deps] - Remove unused Irony (#1685)

Ethereal77 commented:
I could try to include the Assembly Processor project in the main solution, setting the order so every other project builds after it. It could also be set to build first in `Stride.build` I think.

However, I don't know the real reason Silicon / Xenko devs separated the Assembly Processor in its own solution, so don't know if it was to avoid some potential problem when building.

Ethereal77 commented:
In theory, the assembly produced by compiling the Assembly Processor is not locked because it is first copied to `deps`, and then also copied to a temporary directory based on its computed hash.
If you build Stride and MSBuild invokes the Assembly Processor task, that assembly could become locked and unable to be overwritten until you terminate MSBuild or something.
But, if you modify the Assembly Processor, its hash should change, so the new assembly should be copied to a different place and shouldn't be any problem.

Maybe we could ask xen what was the reasoning for keeping it separate.

Eideren committed:
Show useful error message and directions to users lacking requirements for Stride.Core.Tasks

Eideren committed:
Merge remote-tracking branch 'origin/master' into tasks_clarification

Eideren committed:
Revert "Merge remote-tracking branch 'origin/master' into tasks_clarification" (#1701)

This reverts commit 06dc98ab5e742d9ac34c5422403e25cd278e2e00.

Jklawreszuk committed:
Remove native code from stride.core project (#1688)

* Remove native code from stride.core project

Eideren commented:
Sure, @xen2 if you could look into the above two messages.

Eideren committed:
Show useful error message and directions to users lacking requirements for Stride.Core.Tasks

manio143 commented on sources/editor/Stride.GameStudio/Stride.GameStudio.csproj:
```diff
@@ -34,9 +34,9 @@
+    <PackageReference Include="Mono.Cecil" Version="0.11.5" />
     <Reference Include="Stride.Core.AssemblyProcessor">
-      <HintPath Condition="$(TargetFramework.StartsWith('net6'))">..\..\..\deps\AssemblyProcessor\netstandard2.0\Stride.Core.AssemblyProcessor.dll</HintPath>
+      <HintPath>..\..\..\deps\AssemblyProcessor\$(StrideFramework)\Stride.Core.AssemblyProcessor.dll</HintPath>
```
If you're referring to `$(StrideFramework)` here would it make sense to use it in the AssemblyProcessor project as well (after importing the props that defines it)?

Ethereal77 replied:
Yes, maybe that would be a problem. As it is a direct reference to the DLL, it becomes locked when building Game Studio. Or so I think. I remember having read something about avoiding a file lock when building projects in parallel or something.

Maybe if that is replaced with a project reference it'll work?
A project reference would in theory copy the output DLL to the output directory of GameStudio and load it from there, isn't it?
But, about that DLL, the output of Assembly Processor is not just that DLL, it's a different one resulting from packing the Assembly Processor with its dependencies.

Aside: What's the actual advantage of packing the Assembly Processor with its dependencies?

manio143 replied:
I think that may have been a variant of a SingleFilePublish before that was a thing to get a minimal executable to be invoked.
Would be good to understand what is the dependency of the editor on the assembly processor.

Ethereal77 replied:
A little idea I had today. Would there be any problem if GameStudio consumes the AssemblyProcessor as a project reference?

That way, the DLL in `deps/AssemblyProcessor` is only used by the SDK to copy to `%TEMP%` and load the MSBuild task from there, as it does right now. But GameStudio, that uses the AssemblyProcessor directly, would consume the DLL from the project output, MSBuild manages the copy of newer versions, etc., and (with luck) there will be no locks while building the solution.

May it work?

xen2 replied:
Sorry for the delay.
Good idea, I think this approach might work.
One point to check is how it behaves when GameStudio runs a build (it is running its own MSBuild instance), but I think it should be OK.

Eideren committed:
Add dot between major and minor

Jklawreszuk committed:
Separate VisualStudio projects from main solution (#1684)

Jklawreszuk committed:
Fix edge case collision behavior with BoxColliderShape by creating Box2D shape when is2D (#1708)

IXLLEGACYIXL committed:
[GameStudio] Fix sub projects reloading - changing the comparison project related and not UPath related (#1704)

* fix creation and renaming for sub projects

bro-these-are-all-taken committed:
[Rendering] Vulkan Fullscreen Fix (#1709)

* Enabling swapchain recreation in SwapChainGraphicsPresenter.Vulkan.cs so that changes in resolution at runtime don't stop the game from rendering to the screen.

IXLLEGACYIXL committed:
[WIP] Replace NativeStream with BinaryWriter (#1710)

* Rename false Comments of Nativestream to Stream

* Replace any calls to BinaryReader/BinaryWriter

* Remove native stream IO

* remove using directives of deleted IO namespace

* Remove obsolete comment

* replace nativestream comment and block writing with copyto as it allready exists

Ethereal77 committed:
[Rendering] Fix missing Utilities.CopyMemory for other graphics APIs (#1697)

* Fix Direct3D 12 uses of Utilities.CopyMemory

* Also some minor cleanup

* Fix OpenGL uses of Utilities.CopyMemory

* Also some minor cleanup

* Fix Vulkan uses of Utilities.CopyMemory

* Also some minor cleanup

* Add a section to Stride.props where users can set design-time graphics API

* Fix more usages of CopyMemory and ClearMemory

* Also remove some helper methods that copied memory in favor of new built-in ones

* Comment modifications based on feedback

* Fix a deserialization issue introduced in commit 189e96980e96d6180e18a8f7e9466945ff9ecbe8

Eideren committed:
Docs: Readme, update Community Projects link (#1720)

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: add Jklawreszuk as a contributor for code (#1727)

* docs: update README.md [skip ci]

* docs: update .all-contributorsrc [skip ci]

---------

Co-authored-by: allcontributors[bot] <46447321+allcontributors[bot]@users.noreply.github.com>

Jklawreszuk committed:
Change Assimp to Silk.NET.Assimp implementation (#1719)

* Changed Assimp binding to Silk.Net.Assimp

* Converted Stride.Importer.Assimp to a C# library and changed assimp binding to Silk.Net.Assimp
* Converted Stride.Importer.Common to a C# library as this did not contain any native code

allcontributors[bot] committed:
docs: add Ethereal77 as a contributor for code (#1729)

allcontributors[bot] committed:
docs: add tamamutu as a contributor for code (#1731)

allcontributors[bot] committed:
docs: add IXLLEGACYIXL as a contributor for code (#1732)

allcontributors[bot] committed:
docs: add arturoc as a contributor for code (#1733)

allcontributors[bot] committed:
docs: add garychia as a contributor for code (#1734)

allcontributors[bot] committed:
docs: add NicusorN5 as a contributor for code (#1735)

allcontributors[bot] committed:
docs: add mattiascibien as a contributor for code (#1736)

Jklawreszuk committed:
Update README badges (#1737)

Jklawreszuk committed:
[deps] - Remove unused assimp source code

Eideren committed:
[Samples] Bump samples version (#1756)

VaclavElias committed:
Readme.md update - Improving readme based on the community feedback (#1739)

* Removed duplicated badge

* Missing lfs added

arturoc committed:
Fix DX11 shared textures

Right now creating a texture from a native texture with D3D11_RESOURCE_MISC_SHARED_NTHANDLE  will fail unless it is also D3D11_RESOURCE_MISC_SHARED_KEYEDMUTEX. This combination is recomended but not mandatory acording to the specification: https://learn.microsoft.com/en-us/windows/win32/api/d3d11/ne-d3d11-d3d11_resource_misc_flag

Eideren committed:
[System] Fix binary path location when the dlls are run from an unexpected executable

xen2 committed:
Fix VSIX build

xen2 committed:
Assimp: preload assimp library from Ultz.Native.Assimp so that it can be found when actually needed (fixes #1765)

xen2 committed:
Removed UWP from README.md

VaclavElias committed:
[Service] Xml comments improvements (#1768)

Aggror committed:
Update README.md

I build stride source from scratch today but building worked only partially.

I was have SDK 7 installed, but was still missing SDK 6.0.

Also the Git LFS clone command is deprecated and buildin into Git clone

tebjan committed:
[Shader] Adds sRGB and Gamma conversions to ColorUtility.sdsl (#1744)

Co-authored-by: tebjan <tebjan@gmail.com>

Basewq committed:
[Graphics] Fix expected image size for OpenGL textures. (#1786)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

manio143 committed:
[Profiler] Refactor Profiler and submit metrics to .NET Diagnostics API (#1774)

Eideren committed:
[Asset] Show warning message when importing assets that are not inside of the resource folder (#1781)

manio143 committed:
Merge pull request #1746 from Jklawreszuk/assimp-code

[deps] - Remove unused assimp source code

manio143 committed:
Merge pull request #1702 from Eideren/clarification_2

[Build] Show useful error message and directions for users lacking requirements to run Stride.Core.Tasks

Eideren committed:
[Project] Show custom message for users with outdated dotnet version (#1767)

Eideren committed:
[Editor] PropertyGrid, re-implement member sorting by declaration/token (#1790)

Doprez committed:
[Sample] Fix SpriteFont sample, typo in service fetch (#1798)

SVNMLR committed:
[Editor] Add dynamic snapping (#1801)

Implements a dynamic snapping used while holding down a key on manipulating (rotating/moving/scaling) an object/entity

chkw-rks committed:
[Readme] Update logo for dark and light theme (#1587)

Jklawreszuk committed:
[deps] - Remove unused WindowsAPICodePack (#1747)

IXLLEGACYIXL committed:
[Editor] Repair NPE introduced in #1781 (#1804)

Co-authored-by: IXLLEGACYIXL <ixllegacy123@outlook.com>

tebjan committed:
Adds Awesome Stride to readme

VaclavElias committed:
Update README.md - Bullet point fixed in Prerequisites (#1808)

cNoNim committed:
[Editor] Fix some xaml errors (#1822)

Eideren committed:
[Graphics] Dx11: Fix ResizeBuffer when swap chain is under flip model and changing format (#1814)

VaclavElias committed:
[Core] Basic support for F# and VB Project Types (#1821)

SeleDreams committed:
[Editor] Added the ability to copy imported assets automatically to the Resources dir (#1827)

Doprez committed:
[Samples] Add more documentation for beginner tutorial (#1830)

MeharDT committed:
[Editor] Check to see if the .NET 6.0 SDK is Installed When Launching Stride (#1809)

Jklawreszuk committed:
Remove unused MonoDevelop.Debugger (#1748)

Jklawreszuk committed:
[deps] - Remove unused monolinker (#1749)

SVNMLR committed:
[Editor] Add new Theme & UI fixes/refinements (#1807)

MeharDT committed:
[iOS] .NET 6 Template (#1833)

- Updated iOS template .csproj to support .NET Core/6.0, Stride will no longer throw an error when generating iOS projects

cNoNim committed:
Fix compilation of AssemblyProcessor to netstandard2.0

cNoNim committed:
Update Mono.Cecil version to suppress version mismatch

Eideren committed:
[Physics] Pre and Post simulation tick event (#1831)

Eideren committed:
[Editor] Log viewer shows stacktrace by default (#1844)

xen2 committed:
Merge pull request #1840 from cNoNim/fix-assemblyprocessor

Fix compilation of AssemblyProcessor to netstandard2.0

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

JeromyWalsh committed:
Replaced 'xcopy' Exec task with the more universal Copy task to support OS's which do not have xcopy.

manio143 committed:
Merge pull request #1854 from JeromyWalsh/master

Replaced 'xcopy' Exec task with the more universal Copy task to suppo…

Jklawreszuk committed:
[Editor] Translations fix (#1717)

Commit:
feat(theme): improve editor tooltip legibility in Light Steel Blue theme

The Light Steel Blue theme had issues with legibility in editor tooltips, where it used black font on an almost-black background. This change addresses this problem by updating the theme styles.

Fixes #1773

Ethereal77 commented:
Ok, then I may look at try what I wrote in my comment above. Or maybe it's better to merge this and try that in another PR?

xen2 commented:
> Ok, then I may look at try what I wrote in my comment above. Or maybe it's better to merge this and try that in another PR?

Up to you (whatever is more convenient).
If you want to merge this one now, some conflicts need to be fixed.
If not merging now (merging everything at once later), could you make those new changes as a separate/new PR built on top of this one [Ethereal77:assemblyprocessor-fix](https://github.com/Ethereal77/stride/tree/assemblyprocessor-fix)? I think it would make it easier to review only the new changes added on top of this.

manio143 committed:
Merge pull request #1863 from acastrodev/topic/acastrodev/1191

fix(theme): Improve editor tooltip legibility in Light Steel Blue theme

Eideren committed:
[Serialization][Editor] Reference type support on interface fields and properties (#1842)

Eideren committed:
[Core] Fix StringComparison across the project (#1846)

Eideren committed:
[Physics] Change simulation substeping behavior (#1849)

Eideren committed:
[Physics] Add ray test flags (#1850)

Ethereal77 commented:
Of course! Should I rebase this one for pulling the latest changes in `master`? To minimize conflicts, I mean.

JeromyWalsh committed:
[Build] Fix wildcard expansion with AssemblyProcessor + PropertyGroup for environment variables (#1860)

Eideren commented:
Just letting you know that two other PRs were merged related to some of the files you're working on:
https://github.com/stride3d/stride/pull/1860
https://github.com/stride3d/stride/pull/1854
I hope those conflicts aren't too hard to resolve.

parhamgholami committed:
[Editor] Always show directional light arrow (#1865)

Ethereal77 commented:
I know. I've been following them closely. I'll rebase then and see what comes.

manio143 committed:
[Roslyn] Add base setup for a Roslyn integration into Stride.Core (#1847)

* Set up analyzer project to be included in every project

* Create basic structure

* Add Tests project

* Remove generator, fixup tests

* Remove test analyzer

* Add compiler services to unit tests projects

xen2 committed:
Readd VisualStudio.Commands project (they were removed from #1684 but we want to build the Commands project as part of runtime release, even when we don't build/release VSIX package)

xen2 committed:
Fix VSPackage build

MeharDT committed:
[Editor] Fix exception with the Privacy Policy link in the Crash Reporter (#1878)

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

Ethereal77 committed:
Fix performance warnings by using spans where available

Ethereal77 committed:
Minor cleanups

Ethereal77 commented:
I've rebased to latest master and also corrected some performance warnings that were left from a previous change.
Please review.

Eideren commented:
Thanks, tried it out on my end, it does work but there's an issue I want to discuss.

If the `AssemblyProcessor` is not built, opening the solution fails to load a couple of projects:
```
The project 'Stride.Graphics' ran into a problem during the last operation: The expression "[System.IO.File]::ReadAllText(J:\stride\deps\AssemblyProcessor\\netstandard2.0\Stride.Core.AssemblyProcessor.Packed.dll.hash)" cannot be evaluated. 
Could not find file 'J:\stride\deps\AssemblyProcessor\netstandard2.0\Stride.Core.AssemblyProcessor.Packed.dll.hash'.  J:\stride\sources\targets\Stride.Core.targets You may need to reload the solution after fixing the problem. 
```
GameStudio cannot be loaded because of that issue, and building the solution itself fails with a similar error.
Going to the `AssemblyProcessor` solution and building it, then opening the main solution back does fix the issue of course.
I do remember we talked about asking users to manually build the processor before the main solution, did you not find a more elegant solution in the end, or is that just an unrelated issue ?

Ethereal77 commented:
As I said above, I can try to integrate the AssemblyProcessor projects in the main solution. But was deciding if I should do it in this PR or make a new one.
I agree the experience to have to build first the `Stride.AssemblyProcessor.sln` is rough, although it is what I've always done. But for new users it's not evident (and it's not clearly documented) that they should do that.

What should I do?
 1. Continue in this PR with the "experimental" way I described in [this comment](https://github.com/stride3d/stride/pull/1694#discussion_r1246670304)? 
 2. Or should I add the instructions to the `README,md` to build the `Stride.AssemblyProcessor.sln` first, and leave the integration into the main solution for another PR?

xen2 commented:
Sorry if I wasn't clear before:
I was suggesting to do a separate PR on top of this one (approach 1 as you mentioned in https://github.com/stride3d/stride/pull/1694#discussion_r1246670304?). Reason for separate PR is for easier review of that second part only.
However, let's not merge this PR until the second PR is ready. I think lot of people would be confused if need to build another solution (esp. if it's temporary and will be fixed later).
